main/lsp/LSPFileUpdates.cc

```diff
@@ -117,6 +121,9 @@ LSPFileUpdates::fastPathFilesToTypecheck(const core::GlobalState &gs, const LSPC
     int i = -1;
     for (auto &oldFile : gs.getFiles()) {
         i++;
+
+        bool insertedIntoExtraFiles = false;
```

Comment:
Why do we sometimes want to insert early and sometimes we want to insert late? This seems like a bug waiting to happen which is why I'd like to understand the context.

Reply:
Streamlined the logic in that bit
https://github.com/sorbet/sorbet/pull/7573/commits/86f9b15cfa5440549399ae3680ed33adc646cd5a